packages/node/src/integrations/undici/index.ts

```diff
@@ -106,7 +105,7 @@ export class Undici implements Integration {
     let ds: DiagnosticsChannel | undefined;
     try {
       // eslint-disable-next-line @typescript-eslint/no-var-requires
-      ds = dynamicRequire(module, 'diagnostics_channel') as DiagnosticsChannel;
+      ds = require('diagnostics_channel') as DiagnosticsChannel;
```

Comment:
One reason this is there is because `diagnostics_channel` is only available for Node 16+, won't bundling break when bundling for older versions? (Like Node 14).

Reply:
I see it as follows: Every bundler has a built-in escape hatch because `diagnostics_channel` can always be manually externalized. The newer (sane) webpack versions automatically seem to externalize it out of the box, even on older Node.js versions.

Right now `dynamicRequire` seems to break fetch instrumentation on the Next.js canary: https://github.com/getsentry/sentry-javascript/actions/runs/7713179805/job/21023035517

With this change, it succeeds: https://github.com/getsentry/sentry-javascript/actions/runs/7712921252/job/21022104165

I have the soft opinion that this change is the lesser evil but I am not totally sure tbh. People affected would be people who are on old node versions && on old webpack versions. Wdyt?